src/OrchardCore.Modules/OrchardCore.Contents/Views/ContentsAdminList-Create.cshtml

```diff
@@ -1,8 +1,14 @@
+@using Microsoft.AspNetCore.Authorization
 @model ContentOptionsViewModel
+@inject IContentManager ContentManager
+@inject IAuthorizationService AuthorizationService
+
 <div class="form-group">
     <div class="btn-group">
         @if (Model.CreatableTypes.Any())
         {
+            ContentItem contentItem = await ContentManager.NewAsync(Model.CreatableTypes?.First().Value);
+            var hasEditContentPermission = await AuthorizationService.AuthorizeAsync(User, OrchardCore.Contents.Permissions.EditContent, contentItem);
```

Comment:
this check should have already happened in the controller, and it shouldn't be in the `CreatableTypes` list.

If it isn't happening in the controller it should.

It would also have to happen on every type, not just the first one

Reply:
Seems the check was there before I added my changes, I will remove it then

Reply:
The list of CreatableTypes should just not contain the ones you can't create. And even that is a minor issue IMO if that complexifies the code too much (or make it too slow).

Reply:
This code you've added needs to be in the controller, or not at all

Reply:
@deanmarcussen AFAIK when the permission is not applied in the controller we returns `Forbid`, so in this case how can I check the permission If I move it to the controller?

Reply:
See where the list is being created

https://github.com/OrchardCMS/OrchardCore/blob/c8e7ea5cdf74042568f61465a21474338640b6fc/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L126

https://github.com/OrchardCMS/OrchardCore/blob/c8e7ea5cdf74042568f61465a21474338640b6fc/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L163

Reply:
I didn't dig into the code before, it's seems the check is already there, so I remove the one in the view